elasticsearch-transport/README.md

```diff
@@ -208,6 +209,32 @@ You can pass the client any conforming logger implementation:
     log.level = :info
 
     client = Elasticsearch::Client.new logger: log
+### Identifying running tasks with X-Opaque-Id
+
+The X-Opaque-Id header allows to track certain calls, or associate certain tasks with the client that started them ([more on the Elasticsearch docs](https://www.elastic.co/guide/en/elasticsearch/reference/master/tasks.html#_identifying_running_tasks)). To use this feature, you need to set an id for `opaque_id` on the client before each request. Example:
+
+```ruby
+client = Elasticsearch::Client.new
+client.opaque_id = '123456'
+client.search(index: 'myindex', q: 'title:test')
+```
+The search request will include the following HTTP Header:
+```
+X-Opaque-Id: 123456
+```
+
+Please note that `opaque_id` will be set to nil after every request, so you need to set it on the client for every individual request.
+
+You can also set a prefix for X-Opaque-Id when initializing the client. This will be prepended to the id you set before each request. Example:
```

Comment:
Resetting `client.opaque_id` seems too viral. Can we make this side effect free by passing it as part of every client API invocation e.g `client.search()` et all.

Reply:
I thought about that but didn't know how it would work with the setter. I could do `client.search(opaque_id: 'whatever')`. What would the setter do then? Do we need it?

Reply:
The setter on client is then not needed